Code/autopkglib/Unarchiver.py

```diff
@@ -29,6 +29,7 @@
     "tar_bzip2": ["tar.bz2", "tbz"],
     "tar": ["tar"],
     "gzip": ["gzip"],
+    "gunzip": ["gz"],
```

Comment:
Pardon to randomly chime in, but... can this be expanded to be applicable for .xz files as well, since they both initially use gunzip? In my case ([koalaman/shellcheck](https://github.com/koalaman/shellcheck/releases)) that would also require a second pass for un-tar'ing, which I guess can be multiple calls to the same processor...?

Reply:
er, actually @scriptingosx's [using](https://github.com/autopkg/scriptingosx-recipes/blob/e7044730fb62c214f28c949929b1a311314b106c/Shellcheck/Shellcheck.pkg.recipe#L31-L42) a custom processor (and sticking xz as a compiled binary into git AND shelling out to call it...) to perform the second unarchive so I guess that's already an established pattern

Reply:
Since xz isn't built-in, this would require a separate argument variable to define a `decompressor` as @scriptingosx has done.  If this were added here, what would the expected behavior be when there's some sort of mismatch between the defined decompressor and the archive file?

- When `decompressor` is defined and it doesn't match the file type, use the correct one instead?
- When `decompressor` is defined and it doesn't match the file type, fail.

My vote would be for the latter.

Reply:
Seems more complicated than we need to get? `xz` can just be added to the array like there is already a precedent for in this processor, the exact same flags as for gunzip work to decompress it (at least for shellcheck as an example product). The `xz` binary's inclusion in @scriptingosx's repo is neither a good idea nor necessary, as he says in another eclipsed/unnecessary processor for shellcheck, it was added for expediency.

Reply:
OK, got it.  Added `xz`.